### PR TITLE
catch value errors

### PR DESCRIPTION
**Pull request by tallywiesenberg**
Blocked on test mocking. Getting `TypeError: test_rpc_value_errors.<locals>.raise_() takes 0 positional arguments but 1 was given`

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (oraclown) [inline .vscode/settings.json:1]: Add common python & vscode stuff to the gitignore and don't include files like these

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[[source]]
```
Comment (oraclown) [inline Pipfile:1]: Don't need a Pipfile if you have a pyproject.toml & poetry.lock. Delete

File: vars.example.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export TWILIO_AUTH_TOKEN=""
 export TWILIO_ACCOUNT_SID=""
-export TWILIO_FROM="+1231231234"
-export ALERT_RECIPIENTS="+1231231234,+1231231234,+1231231234"
+export TWILIO_FROM=""
+export ALERT_RECIPIENTS=""
```
Comment (oraclown) [inline vars.example.sh:4]: Why take out the examples?

**Review by oraclown — CHANGES_REQUESTED** [review]
thanks @tallywiesenberg

File: vars.example.sh (hunk 3 above)
Comment (tallywiesenberg) [reply vars.example.sh:4]: ah that was an accident

**Comment by tallywiesenberg** [comment]
Closes #19 #20

File: src/tellor_disputables/data.py [hunk 4]
```diff
@@ -85,24 +84,29 @@ async def log_loop(web3: Web3, addr: str) -> list[tuple[int, Any]]:
...
+            txhash = event["transactionHash"]
 
-    return unique_events_lis
+            if txhash not in unique_events:
+                unique_events[txhash] = event
+                unique_events_list.append((web3.eth.chain_id, event))
+
+        return unique_events_list
+
+    except ValueError as e:
```
Comment (oraclown) [inline src/tellor_disputables/data.py:100]: Exception should wrap the specific line where the error could be generated. Else it's not clear where the error is coming from for future readers/debuggers.

File: src/tellor_disputables/data.py [hunk 5]
```diff
@@ -85,24 +84,29 @@ async def log_loop(web3: Web3, addr: str) -> list[tuple[int, Any]]:
...
+        msg = str(e)
+        if "unknown block" in msg:
+            print("waiting for new blocks")
```
Comment (oraclown) [inline src/tellor_disputables/data.py:103]: Should probs set up logging to a files similar to `telliot_core`/telliot_feeds`, so @spuddyminer can link txt files in the bug issues

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks @tallywiesenberg ! Left some comments / requests

File: src/tellor_disputables/data.py (hunk 5 above)
Comment (tallywiesenberg) [reply src/tellor_disputables/data.py:103]: Referenced #24

File: src/tellor_disputables/data.py (hunk 4 above)
Comment (tallywiesenberg) [reply src/tellor_disputables/data.py:100]: good catch

**Review by oraclown — APPROVED** [review]

**Comment by oraclown** [comment]
Thanks @tallywiesenberg !

**Pull request merged**